java-frontend/src/main/java/org/sonar/java/AbstractJavaClasspath.java

```diff
@@ -102,6 +103,10 @@ protected boolean hasJavaSources() {
     return fs.hasFiles(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)));
   }
 
+  protected boolean hasMoreThanOneJavaFile() {
+    return Iterables.size(fs.files(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)))) > 1;
```

Comment:
What about test binaries (for `org.sonar.java.JavaTestClasspath`)? Should we also fail if `sonar.java.test.binaries` is not provided when there is some test sources or we only issue a warning for this case?